### ISSUE TITLE
NotConnected: Transport endpoint is not connected (os error 107)

### ISSUE THREAD

**Issue by LeonardGreulich** [denoland/deno#8499]
Hey,

some HTTP server errors (e.g., issue https://github.com/denoland/deno/issues/6319) have been fixed by this PR just recently https://github.com/denoland/deno/pull/8365. However, now I get the following error after running an HTTP server behind nginx for a couple of hours on Ubuntu:

```JavaScript
error: Uncaught (in promise) NotConnected: Transport endpoint is not connected (os error 107)
    at processResponse (core.js:223:11)
    at Object.jsonOpAsync (core.js:240:12)
    at async Listener.accept (deno:cli/rt/30_net.js:117:19)
    at async Server.acceptConnAndIterateHttpRequests (server.ts:223:14)
    at async MuxAsyncIterator.callIteratorNext (mux_async_iterator.ts:30:31)
```

@lideming do you have an idea what this could cause?

**Comment by lideming** [comment]
I am working on a fix.

**Comment by LeonardGreulich** [comment]
Thank you!

**Comment by lideming** [comment]
@bartlomieju 
I am not sure why there is an error 107 (i.e. Linux error ENOTCONN), which isn't mentioned in [Linux maunal `accept(2)`](https://man7.org/linux/man-pages/man2/accept.2.html) but is in [`getpeername(2)`](https://man7.org/linux/man-pages/man2/getpeername.2.html). I guess the runtime has called `getpeername()` somewhere just after a client closed the connection.

If PR #8554 is landed, the server should continue to work after encountering such error.

**Comment by timonson** [comment]
Any news on this?  Or at least, why the PR is not accepted, yet? It has not been fixed in v1.6.3 yet.
